### ISSUE TITLE
Unittest of dopamine synapse model

### ISSUE THREAD

**Issue by suku248** [nest/nest-simulator#820]
There are only manual tests of `stdp_dopa_connection` and they are outdated and will be removed with #774. A unittest of this synapse model is required.

**Comment by terhorstd** [comment]
@abigailm is there progress on this one?

**Comment by heplesser** [comment]
Model should be re-created using NESTML, including tests.

**Comment by jougs** [comment]
@clinssen: Whats your take on @heplesser's [comment](https://github.com/nest/nest-simulator/issues/820#issuecomment-481217978)? Would this kind of third-factor plasticity be currently feasible in NESTML? Thanks!

**Comment by heplesser** [comment]
To expand a little on my comment from 1.5 years ago: If this model including tests can be built with NESTML in the near future, that would be a way to go. If not, we need to write tests for the model as is.

**Comment by clinssen** [comment]
Me and @YounesBouhadjar will write a unit test that:

- generates random spike times: pre, post and dopa (but will also test some edge-case predefined times, such as pre and post spikes occurring at the exact same time)
- Runs a NEST simulation with three nodes and a volume transmitter with these spike times, obtain weight recording;
- Runs a Python sim with these spike times
	- precise integration to the next spike (whichever of the three)
	- update values w, c, n using exponential solution of the ODE
	- update values c or n depending on arriving spike
- Compares Python and NEST-obtained timeseries of w

**Comment by jougs** [comment]
@clinssen, @YounesBouhadjar: are there any news on this?

**Comment by clinssen** [comment]
This was pre-empted by some other projects and queries, such as #1840, a NESTML active dendrite tutorial review, and the triplet synapse. As the latter is now working and third-factor plasticity rules are next on the NESTML agenda, we expect to pick this up again within the next few weeks!

**Comment by jougs** [comment]
Thanks for the update and no worries. I was just going through the stale issues in the [automated testing project](https://github.com/nest/nest-simulator/projects/5).

**Comment by jessica-mitchell** [comment]
@heplesser @clinssen is this issue still relevant?

**Comment by heplesser** [comment]
@jessica-mitchell Yes. If we have a feature, we need tests for it. It think we should proceed to write the tests as pytests for NEST. 

@abigailm Would you know someone who could write these tests?

**Comment by heplesser** [comment]
> Ok I just saw that there was a merged PR regarding the unit test for STDP dopamine, and a few tests that reference it, so wasnt sure if it was already dealt with

Do you have the number of that PR? If we have test in place, we might not need more and could close this issue indeed.

**Comment by heplesser** [comment]
#1840 tests plain stdp, but not stdp_dopamine_synapse. I have searched through our testsuite and have not found any test for the correct dynamics of the dopamine synapse.
